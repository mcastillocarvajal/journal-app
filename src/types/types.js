

export const types = {

    login: '[Auth] Login',
    logout: '[Auth] Logout',

    uiSetError: '[UI] Set Error',
    uiRemoveError: '[UI] Remove Error',

    uiStartLoading: '[UI] Start Loading',
    uiFinishLoading: '[UI] Finish Loading',

    notesNewEntry: '[Notes] New Note',
    notesActive: '[Notes] Set Active Note',
    notesLoad: '[Notes] Load Note',
    notesUpdated: '[Notes] Updated Note',
    notesFileUrl: '[Notes] Updated ImageUrl',
    notesDelete: '[Notes] Delete Note',
    notesLogoutCleaning: '[Notes] Logout Cleaning'
}