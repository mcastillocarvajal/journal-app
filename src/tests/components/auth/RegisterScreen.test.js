import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';

import { RegisterScreen } from '../../../components/auth/RegisterScreen';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
}
let store = mockStore( initState);

const wrapper = mount( 
    <Provider store={store} >
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
);

describe('test for <RegisterScreen />', () => {

    test('should show match snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should dispatch the action', () => {
        
        const email = wrapper.find('input[name="email"]');
        
        email.simulate('change', {
            target: {
                value: '',
                name: 'email'
            }
        })

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        const actions = store.getActions();

        expect( actions[0] ).toEqual({
            type: types.uiSetError,
            payload: 'Name is required'
        })
    })
    
    test('should show alert with msg error', () => {

        const initState = {
            auth: {},
            ui: {
                loading: false,
                msgError: 'Email is not valid'
            }
        }
        const store = mockStore( initState);

        const wrapper = mount( 
            <Provider store={store} >
                <MemoryRouter>
                    <RegisterScreen />
                </MemoryRouter>
            </Provider>
        );

        expect( wrapper.find('.auth__alert-error').exists() ).toBe( true );

        expect( wrapper.find('.auth__alert-error').text().trim() ).toBe( initState.ui.msgError );
    })
    
})
