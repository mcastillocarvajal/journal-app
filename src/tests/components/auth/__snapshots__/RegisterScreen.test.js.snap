// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <RegisterScreen /> should show match snapshot 1`] = `
<div
  className="animate__animated animate__fadeIn animate__faster"
>
  <h3
    className="auth__title"
  >
    Register
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block mb-5"
      disabled={false}
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>
</div>
`;
